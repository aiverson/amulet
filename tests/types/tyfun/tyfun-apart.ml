type dict 'c = Dict : 'c => dict 'c
type 'p :- 'q = Sub of ('p => unit -> dict 'q)

type function discrim 'a 'b begin
  discrim 'a 'a = int
  discrim 'a 'b = string
end

class trivial begin end
instance trivial begin end

let foo : forall 'a 'b. 'a ~ 'b :- (discrim 'a 'b ~ int) = Sub (fun _ -> Dict)
let no : forall 'a 'b. trivial :- (discrim 'a 'b ~ string) = Sub (fun _ -> Dict)
