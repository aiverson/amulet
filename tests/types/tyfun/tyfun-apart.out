tyfun-apart.ml[13:76 ..13:79]: error
  No instance for discrim 'a 'b ~ string arising from use of the expression
   │ 
13 │ let no : forall 'a 'b. trivial :- (discrim 'a 'b ~ string) = Sub (fun _ -> Dict)
   │                                                                            ^^^^
